adapta/utils/decorators/_logging.py

```diff
@@ -0,0 +1,53 @@
+""" Module for common decorator methods. """
+from functools import wraps
+
+from adapta.logs import SemanticLogger
+from adapta.logs.models import LogLevel
+from adapta.metrics import MetricsProvider
+from adapta.utils._common import operation_time
+
+
+def run_time_metrics(metric_name: str, tag_function_name: bool = False, reporting_level: LogLevel = LogLevel.DEBUG):
```

Comment:
Maybe we could use `log_level` instead of `reporting_level` like we do other places?